backend/models/assets.py

```diff
@@ -39,22 +41,18 @@ def download_path(self) -> str:
 class RomAsset(BaseAsset):
     __abstract__ = True
 
-    rom_id = Column(
-        Integer(), ForeignKey("roms.id", ondelete="CASCADE"), nullable=False
-    )
-    user_id = Column(
-        Integer(), ForeignKey("users.id", ondelete="CASCADE"), nullable=False
-    )
+    rom_id: Mapped[int] = mapped_column(ForeignKey("roms.id", ondelete="CASCADE"))
+    user_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"))
 
 
 class Save(RomAsset):
     __tablename__ = "saves"
     __table_args__ = {"extend_existing": True}
 
-    emulator = Column(String(length=50), nullable=True)
+    emulator: Mapped[str | None] = mapped_column(String(length=50))
```

Comment:
don't we have to mark this `nullable=True`?

Reply:
We don't! The typehint takes care of that, as explained [in step four](https://docs.sqlalchemy.org/en/20/changelog/whatsnew_20.html#step-four-remove-orm-mapped-column-directives-where-no-longer-needed), so `Mapped[str]` implicitly sets `nullable=False`, while `Mapped[str | None]` sets `nullable=True`.